pkg/imagefilter/imagefilter.go

```diff
@@ -0,0 +1,90 @@
+package imagefilter
+
+import (
+	"fmt"
+
+	"github.com/osbuild/images/pkg/distro"
+	"github.com/osbuild/images/pkg/distrofactory"
+	"github.com/osbuild/images/pkg/distrosort"
+)
+
+type DistroLister interface {
+	ListDistros() []string
+}
+
+// Result contains a result from a imagefilter.Filter run
+type Result struct {
+	Distro  distro.Distro
+	Arch    distro.Arch
```

Comment:
One question here is if we should include non-native arches or provide a way to not look for them (maybe even not list them by default) as we cannot do cross-arch builds for classic rpm based distros yet (because we would need to download a target-arch rpm first to construct the target-arch buildroot). OTOH we could also leave this to the frontend/formatter to just filter out the undesired results (which is what I'm leaning towards right now)